### PR TITLE
Arun emp status b35

### PR DESCRIPTION
**Pull request by Arun1919V**
kindly review my automation of Employment Status page

### CODE REVIEW

File: src/test/java/pages/Employment_Status.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    public boolean saveNewES(String eStatus) {
+        click(By.xpath(ESAdd));
+        setText(By.xpath(ESAddStatus), eStatus);
+        if (getText(By.xpath(ESAlredyExisting)).contains("Already exists")) {
+            click(By.xpath(ESCancelData));
+        } else {
+            click(By.xpath(ESSaveData));
+
+            isElementVisible(By.cssSelector(ESTable));
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java]: Check the visibility of "Successfully saved" pop-up.

File: src/test/java/pages/Employment_Status.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                String txtEmpStatus = EmpStatus.getText();
+                System.out.println(txtEmpStatus);
+
+            }
+            Boolean match = ESlist.stream().map(s -> s.getText()).anyMatch(s -> s.equalsIgnoreCase(eStatus));
+
+
+
+        }
+        return true;
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java]: This method is always returning true.

File: src/test/java/pages/Employment_Status.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+    }
+
+    public String success() {
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java:75]: Check the logic behind this method.

File: src/test/java/pages/Employment_Status.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+        return getText(By.xpath(ESSave));
+    }
+
+    public boolean EditEmpStatus() {
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java]: Not complete

File: src/test/java/pages/Employment_Status.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        sleep(3000);
+        click(By.xpath(ESEdit));
+        /*List<WebElement> ESL = driver.findElements(By.xpath(ES));
+        for(int i = 0; i< ESL.size();i++) {
+            WebElement Edit = ESL.get(i);
+            List<WebElement> EB = driver.findElements(By.xpath(EditButton));
+            for (int j = 1; j < EB.size(); j++) {
+                //WebElement xy = EB.get(j);
+                String EH = Edit.getText();
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java]: Traverse through the table, find the row with input text and do edit.

Assertion should pass only if the employment status you have added is present on the table.

File: src/test/java/pages/Employment_Status.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        List<WebElement> ESL = driver.findElements(By.xpath(ES));
+        for(int i = 0; i< ESL.size();i++){
+            WebElement Trash = ESL.get(i);
+            List<WebElement> TB = driver.findElements(By.xpath(TrashButton));
+            for(int j=0; j<TB.size();j++) {
+                //WebElement xy = TB.get(j);
+                String EX = Trash.getText();
+                if (EX.equalsIgnoreCase(deleteStatus)) {
+                    click(By.xpath(TrashButton));
+                    sleep(2000);
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java]: Recheck the logic

File: src/test/java/pages/Employment_Status.java [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+                    click(By.xpath(ESDeleteYesDelete));
+                    break;
+                }
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java:115]: Verify only the given status is deleted

File: src/test/java/pages/Employment_Status.java [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+        click(By.xpath(ESDeleteSelected));
+        click(By.xpath(ESDeleteYesDelete));
+        isElementVisible(By.cssSelector(ESTable));
+        System.out.println(ESTable);
+        for (WebElement EmpStatus : ESlist) {
+            String txtEmpStatus = EmpStatus.getText();
+            System.out.println(txtEmpStatus);
+
+        }
+        return null;
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java]: Why this method is returning null?

Add employment statuses which you're going to delete as method parameters and make sure only those are getting deleted from table.

File: src/test/java/pages/Employment_Status.java [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+        }
+        return true;
+    }
+
+
+    public String deleteRandomMultipleEmpStatus() {
+        sleep(3000);
+        click(By.xpath(ESMultiDelete1));
+        click(By.xpath(ESMultiDelete2));
+        click(By.xpath(ESDeleteSelected));
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java:125]: Change the logic. Traverse through the table.

File: src/test/java/pages/Employment_Status.java [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+        return null;
+    }
+
+
+    public String countofRecord() {
+        sleep(3000);
+        return getText(By.xpath(ESRecord));
+
+    }
+}
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java:138]: Rename variable accordingly.

File: src/test/java/library/PageBase.java [hunk 11]
```diff
@@ -119,6 +119,23 @@ protected void setText(By by, String text, int ... retries) {
             }
         }
     }
+    protected Void setText1(By by,String text){
+        WebElement we = new WebDriverWait(driver, Duration.ofSeconds(WAIT_TIME))
+                .until(ExpectedConditions
+                        .presenceOfElementLocated(by));
```
Comment (xploreejav) [inline src/test/java/library/PageBase.java:125]: Recheck the purpose of setText1, if it's needed rename the method.

File: src/test/java/tests/Employment_StatusTests.java [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Test(priority = 1)
+    public void EditEmpStatus() throws InterruptedException {
+        LoginPage loginPage = new LoginPage(driver);
+        loginPage.login("Admin", "admin123", true, null);
+        HeaderPage headerPage = new HeaderPage(driver);
+        headerPage.selectMenu(MenuOptions.EMPLOYMENT_STATUS);
+        Employment_Status EditStatus = new Employment_Status(driver);
+        Assert.assertEquals(EditStatus.EditEmpStatus(),true);
+        driver.findElement(By.xpath("(//input[contains(@class,'oxd-input oxd-input')])[2]")).sendKeys("Operation Manager");
```
Comment (xploreejav) [inline src/test/java/tests/Employment_StatusTests.java]: Remove line 40,41,42.These should be done inside the method EditEmpStatus.

File: src/test/java/pages/Employment_Status.java [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+
+    WebDriver driver;
+
+    private String ESAdd = "//button[text()=' Add ']";
+    private String ESAddStatus = "//div[@class='oxd-input-group oxd-input-field-bottom-space']//child::input";
+    private String ESSaveData = "//button[text()=' Save ']";
+    private String ESCancelData = "//button[text()=' Cancel ']";
+    private String ESAlredyExisting = "//div[@class='oxd-form-row']/div";
+    private String ESTable = ".oxd-table-body";
+    private String ESEdit = "(//button[@class='oxd-icon-button oxd-table-cell-action-space'])[2]";
```
Comment (xploreejav) [inline src/test/java/pages/Employment_Status.java]: Recheck xpaths

File: src/test/java/tests/Employment_StatusTests.java [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    @Test(priority = 1)
+    public void EditEmpStatus() throws InterruptedException {
+        LoginPage loginPage = new LoginPage(driver);
+        loginPage.login("Admin", "admin123", true, null);
+        HeaderPage headerPage = new HeaderPage(driver);
+        headerPage.selectMenu(MenuOptions.EMPLOYMENT_STATUS);
+        Employment_Status EditStatus = new Employment_Status(driver);
+        Assert.assertEquals(EditStatus.EditEmpStatus(),true);
```
Comment (xploreejav) [inline src/test/java/tests/Employment_StatusTests.java]: Add employment status you are going to edit as parameter

File: src/test/java/tests/Employment_StatusTests.java [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+        Employment_Status DeleteMultiStat = new Employment_Status(driver);
+        Assert.assertEquals(DeleteMultiStat.deleteRandomMultipleEmpStatus(),null);
+
+    }
+    @Test(priority = 4)
+    public void CountOfRecords(){
+        LoginPage loginPage = new LoginPage(driver);
+        loginPage.login("Admin", "admin123", true, null);
+        HeaderPage headerPage = new HeaderPage(driver);
+        headerPage.selectMenu(MenuOptions.EMPLOYMENT_STATUS);
```
Comment (xploreejav) [inline src/test/java/tests/Employment_StatusTests.java:69]: Recheck purpose of this test

File: testngEmpStatus.xml [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+                <include name="EditEmpStatus"/>
+                <include name="DeleteEmpStatus"/>
+                <include name="DeleteMultiEmpStats"/>
+                <include name="CountOfRecords"/>
+            </methods>
+
+        </class>
+        </classes>
+    </test>
+</suite>
```
Comment (xploreejav) [inline testngEmpStatus.xml:17]: Change the branch name. Kindly use the format: Yourname-Pagename-Jiraid

File: src/test/java/pages/Employment_Status.java (hunk 2 above)
Comment (Arun1919V) [reply src/test/java/pages/Employment_Status.java]: i was performing assertion inside the test

**Review by xploreejav — CHANGES_REQUESTED** [review]
Kindly check all the comments.

File: src/test/java/pages/Employment_Status.java (hunk 3 above)
Comment (Arun1919V) [reply src/test/java/pages/Employment_Status.java:75]: i was not using this method

File: src/test/java/pages/Employment_Status.java (hunk 4 above)
Comment (Arun1919V) [reply src/test/java/pages/Employment_Status.java]: I was not using it

**Review by thanexplore — CHANGES_REQUESTED** [review]
.idea folder should not be committed
unused methods should not be added to the branch for commit
kindly do another commit with the changes requested

**Review by Arun1919V — COMMENTED** [review]
Kindly review my Employement_Status page

**Review by thanexplore — CHANGES_REQUESTED** [review]
Kindly remove the .idea folder files, target folder files

**Comment by thanexplore** [comment]
this PR not be considered as with another branch completed the project

**Pull request closed without merging**